src/System.Runtime.Intrinsics.X86/src/System/Runtime/Intrinsics/X86/Avx2.cs

```diff
@@ -0,0 +1,725 @@
+    /// AVX2 class provides access to 256-bit SIMD instructions
+    /// </para>
+
+    public static class Avx2
+    {
+        public static bool IsSupported { get; }
+        
+        // __m256i _mm256_abs_epi8 (__m256i a)
+        public static Vector256<byte> Abs(Vector256<sbyte> value) { throw new NotImplementedException(); }
+        // __m256i _mm256_abs_epi16 (__m256i a)
+        public static Vector256<ushort> Abs(Vector256<short> value) { throw new NotImplementedException(); }
+        // __m256i _mm256_abs_epi32 (__m256i a)
+        public static Vector256<uint> Abs(Vector256<int> value) { throw new NotImplementedException(); }
+
+        // __m256i _mm256_add_epi8 (__m256i a, __m256i b)
+        public static Vector256<sbyte> Add(Vector256<sbyte> left, Vector256<sbyte> right) { throw new NotImplementedException(); }
+        // __m256i _mm256_add_epi8 (__m256i a, __m256i b)
+        public static Vector256<byte> Add(Vector256<byte> left, Vector256<byte> right) { throw new NotImplementedException(); }
+        // __m256i _mm256_add_epi16 (__m256i a, __m256i b)
+        public static Vector256<short> Add(Vector256<short> left, Vector256<short> right) { throw new NotImplementedException(); }
+        // __m256i _mm256_add_epi16 (__m256i a, __m256i b)
+        public static Vector256<ushort> Add(Vector256<ushort> left, Vector256<ushort> right) { throw new NotImplementedException(); }
+        // __m256i _mm256_add_epi32 (__m256i a, __m256i b)
+        public static Vector256<int> Add(Vector256<int> left, Vector256<int> right) { throw new NotImplementedException(); }
+        // __m256i _mm256_add_epi32 (__m256i a, __m256i b)
+        public static Vector256<uint> Add(Vector256<uint> left, Vector256<uint> right) { throw new NotImplementedException(); }
+        // __m256i _mm256_add_epi64 (__m256i a, __m256i b)
+        public static Vector256<long> Add(Vector256<long> left, Vector256<long> right) { throw new NotImplementedException(); }
+        // __m256i _mm256_add_epi64 (__m256i a, __m256i b)
+        public static Vector256<ulong> Add(Vector256<ulong> left, Vector256<ulong> right) { throw new NotImplementedException(); }
+
+        // __m256i _mm256_adds_epi8 (__m256i a, __m256i b)
+        public static Vector256<sbyte> AddSaturation(Vector256<sbyte> left, Vector256<sbyte> right) { throw new NotImplementedException(); }
+        // __m256i _mm256_adds_epu8 (__m256i a, __m256i b)
+        public static Vector256<byte> AddSaturation(Vector256<byte> left, Vector256<byte> right) { throw new NotImplementedException(); }
+        // __m256i _mm256_adds_epi16 (__m256i a, __m256i b)
+        public static Vector256<short> AddSaturation(Vector256<short> left, Vector256<short> right) { throw new NotImplementedException(); }
+        // __m256i _mm256_adds_epu16 (__m256i a, __m256i b)
+        public static Vector256<ushort> AddSaturation(Vector256<ushort> left, Vector256<ushort> right) { throw new NotImplementedException(); }
+
```

Comment:
Perhaps `AddSaturate` or `AddWithSaturation` could better communicate action performed by method (it does not add Saturation).